Minimal restructure to demo Concept/How To/Reference/Tutorial approach

MadelineAu opened the pull request:
@wesfloyd - put together this draft PR to show how a Concept/HowTo/Reference/Tutorial restructure would look.  It's very minimal so far - only moves and renames topics.  I wanted to get your thoughts before going any further and putting redirects in place. 

The diffs won't be great for seeing what I'm trying to do - you'll probably need to pull the branch and build locally.

wesfloyd commented:
<img width="411" alt="image" src="https://github.com/user-attachments/assets/bbb12b59-374f-4a9c-bb3a-779245cd6982" />

wesfloyd commented:
@MadelineAu  Overall I'm a fan of the new organization - it feels very reasonably and logically organized. I don't see any red flags and in general I want to defer to you preference where possible.

It might be good to double check with Nader in case he'd like to weigh in as a next step? Some of the current design decisions were with his guidance in the interests of new developers.

Tactical feedback:
- I needed to fix a few URLs in order for it to build properly. https://github.com/Layr-Labs/eigenlayer-docs/pull/420/commits/f7c9d9ab9c2ff6567334708801d4ac7d07b5825b
- If/when we move forward with these changes - we may want to create new redirects from the old links to the new links. Otherwise external links used in many places would lead to 404.
- Maybe we move Security to "Reference" or a lower section since it hasn't been reviewed in 6 months'ish and is less necessary for first time builders?

dabit3 commented:
A couple of thoughts:

1. I really like this structure, I think it's definitely more intuitive 
2. I wonder if the change in language from "AVS Dashboard Onboarding" to "Implement AVS Marketplace Interfaces" could be confusing as most of the inbound we have for this documentation usually revolves around wanting to be listed on the dashboard
3. From my understanding, the quickstart guide is one of the most useful pieces of documentation we have, I wonder if we can have it somewhere more prominent

MadelineAu commented:
Thanks for the feedback @wesfloyd and @dabit3.  Addressed as follows: 
1. Created (and fixed...) redirects. 
2. Moved Security page to Reference. 
3. Updated to Onboard to AVS Dashboard 
4. Moved Quickstart in to "How To -> Get Started".  Could consider breaking the framework to add a 'get started' section above 'Concepts' if that makes sense.

wesfloyd reviewed (approved):
lgtm
